onnxruntime/core/providers/webnn/builders/impl/einsum_op_builder.cc

```diff
@@ -0,0 +1,598 @@
+                                              const logging::Logger& logger) const {
+  const auto& input_defs = node.InputDefs();
+  const auto& initializers(model_builder.GetInitializerTensors());
+  emscripten::val output = emscripten::val::object();
+
+  NodeAttrHelper helper(node);
+  const auto equation = helper.Get("equation", std::string(" "));
+
+  std::vector<uint32_t> m_label_indices;
+  std::vector<Component> m_components;
+  std::vector<uint32_t> m_output_dimensions;
+  uint32_t num_labels;
+  ORT_RETURN_IF_NOT(ParseEquationComponents(initializers, node, equation, m_label_indices,
+                                            m_components, m_output_dimensions, num_labels, logger),
+                    "Error parsing equation components.");
+
+  RecognizedOperatorType recognized_operator_type = DetermineRecognizedOperatorType(m_label_indices, m_components,
+                                                                                    m_output_dimensions);
+
+  switch (recognized_operator_type) {
+    case RecognizedOperatorType::Multiply: {
+      const size_t a_idx = 0, b_idx = 1;
+      emscripten::val a = model_builder.GetOperand(node.InputDefs()[a_idx]->Name());
+      emscripten::val b = model_builder.GetOperand(node.InputDefs()[b_idx]->Name());
+      output = model_builder.GetBuilder().call<emscripten::val>("mul", a, b);
+    } break;
+    case RecognizedOperatorType::ReduceSum: {
+      auto kept_axes = m_components.back().GetLabels(m_label_indices);
+      assert(kept_axes.size() <= 1);
+      std::vector<uint32_t> reduced_axes;
+      uint32_t kept_axes_mask = 0;
+      for (auto axis : kept_axes) {
+        kept_axes_mask |= (1 << axis);
+      }
+      std::vector<int64_t> input_shape;
+      ORT_RETURN_IF_NOT(GetShape(*input_defs[0], input_shape, logger), "Cannot get shape");
+      for (uint32_t axis = 0, axis_count = static_cast<uint32_t>(input_shape.size()); axis < axis_count; ++axis) {
+        if (~kept_axes_mask & (1 << axis)) {
+          reduced_axes.push_back(axis);
+        }
+      }
+
+      emscripten::val input = model_builder.GetOperand(node.InputDefs()[0]->Name());
+      emscripten::val options = emscripten::val::object();
+      options.set("keepDimensions", false);
+
+      const auto input_rank = input_shape.size();
+      std::vector<int32_t> axes_data;
+      std::transform(
+          reduced_axes.begin(), reduced_axes.end(), std::back_inserter(axes_data),
+          [input_rank](int64_t axis) -> int32_t { return SafeInt<int32_t>(HandleNegativeAxis(axis, input_rank)); });
+      options.set("axes", emscripten::val::array(axes_data));
+
+      output = model_builder.GetBuilder().call<emscripten::val>("reduceSum", input, options);
+    } break;
+    case RecognizedOperatorType::Transpose: {
+      emscripten::val input = model_builder.GetOperand(node.InputDefs()[0]->Name());
+      // Transpose via input strides. The output tensor is not strided.
```

Comment:
This comment doesn't make sense in the current context, since strides are not used in the WebNN EP.